File: Rendering/Test One - Color Changing Circles/pmma test.py
import pmma, time, math, random
from typing import List

display = pmma.Display()
display.create([1280, 720], fullscreen=False, vsync=True)

shapes: List[pmma.Shapes2D.RadialPolygon] = []
for _ in range(100_000):
    shape = pmma.Shapes2D.RadialPolygon()
    shape.shape_center.generate_from_random()
    shape.shape_color.generate_from_random(generate_alpha=False)
    shape.set_radius(50)
    shapes.append(shape)

total = []
n = 0
while True:
    s = time.perf_counter()
    display.clear()

    for shape in shapes:
        shape.render()
        shape.shape_color.generate_from_random(generate_alpha=False)

    display.refresh(limit_refresh_rate=False)
    e = time.perf_counter()
    n += 1

    if n > 10:
        total.append(1/(e-s))
        print(f"Average FPS: {sum(total)/len(total)} ({len(total)})")

    if n > 65:
        break

File: Color And Position Changing 100,000 Circles/PMMA_Test.cpp
#include <string>
#include <vector>
#include <cstdint>

#include "PMMA.hpp"

#include "PMMA_Test.h"

#ifndef PMMA_INSTALL_DIR
    #define PMMA_INSTALL_DIR "extern/pmma" // Fallback default
#endif

void PMMA_Test() {
    std::string path = PMMA_INSTALL_DIR;

    PMMA::Initialize(path);

    PMMA::Display* display = new PMMA::Display();
    std::uint16_t size[2] = { 1280, 720 };
    display->Create(size, { .Vsync = false });

    std::vector<PMMA::Rendering::TwoD::Shapes::Circle> shapes(100'000);

    for (int i = 0; i < 100'000; i++) {
        shapes[i].ShapeCenter.GenerateFromRandom();
        shapes[i].Color.GenerateFromRandom();
        shapes[i].ShapeSize.SetRadius(50);
    };

    while (PMMA::General::IsApplicationRunning()) {
        display->Clear();

        for (auto& shape : shapes) {
            shape.Color.GenerateFromRandom();
            shape.Render();
        }

        display->Refresh({ .LimitRefreshRate = false });
    }

    PMMA::Uninitialize();
}